### PR TITLE
MySQL Grammar Enhancements

### PR DESCRIPTION
**Pull request by Naros**

### CODE REVIEW

**Comment by Naros** [comment]
It appears the build is failing due to a TSQL parser failure.  I have added PR https://github.com/antlr/grammars-v4/pull/1411 to address that separately.

File: mysql/MySqlParser.g4 [hunk 1]
```diff
@@ -1334,7 +1340,7 @@ blockStatement
         (declareCondition SEMI)*
         (declareCursor SEMI)*
         (declareHandler SEMI)*
-        procedureSqlStatement+
+        (procedureSqlStatement)*
```
Comment (KvanTTT) [inline mysql/MySqlParser.g4]: Maybe remove redundant parentheses?

File: mysql/MySqlLexer.g4 [hunk 2]
```diff
@@ -1157,7 +1166,7 @@ fragment CHARSET_NAME:               ARMSCII8 | ASCII | BIG5 | BINARY | CP1250
                                      | UCS2 | UJIS | UTF16 | UTF16LE | UTF32 
                                      | UTF8 | UTF8MB4;
 
-fragment EXPONENT_NUM_PART:          'E' '-'? DEC_DIGIT+;
+fragment EXPONENT_NUM_PART:          'E' [\-+]? DEC_DIGIT+;
```
Comment (KvanTTT) [inline mysql/MySqlLexer.g4]: Not necessary to escape minus `-` here.

**Comment by Naros** [comment]
@KvanTTT, rebased on master since you merged the TSQL fix and made the suggested changes.

**Pull request merged**